core/kernel/persistence/smoothsql/class.Class.php

```diff
@@ -486,11 +486,14 @@ public function getFilteredQuery(core_kernel_classes_Class $resource, $propertyF
         $limit = (isset($options['limit']) === false) ? 0 : $options['limit'];
         $order = (isset($options['order']) === false) ? '' : $options['order'];
         $orderdir = (isset($options['orderdir']) === false) ? 'ASC' : $options['orderdir'];
-        
-        $search = \oat\oatbox\service\ServiceManager::getServiceManager()->get('generis/complexSearch');
-        $query = $search->getQuery($this->getModel(), $rdftypes, $propertyFilters, $and, $like, $lang, $offset, $limit, $order, $orderdir);
-        
-        //$query = core_kernel_persistence_smoothsql_Utils::buildFilterQuery($this->getModel(), $rdftypes, $propertyFilters, $and, $like, $lang, $offset, $limit, $order, $orderdir);
+        /* @var $serviceManager \oat\oatbox\service\ServiceManager */
+        $serviceManager = \oat\oatbox\service\ServiceManager::getServiceManager();
+        if($serviceManager->has('generis/complexSearch')) {
+            $search = $serviceManager->get('generis/complexSearch');
+            $query = $search->getQuery($this->getModel(), $rdftypes, $propertyFilters, $and, $like, $lang, $offset, $limit, $order, $orderdir);
+        } else {
+            $query = core_kernel_persistence_smoothsql_Utils::buildFilterQuery($this->getModel(), $rdftypes, $propertyFilters, $and, $like, $lang, $offset, $limit, $order, $orderdir);
```

Comment:
After a long discussion with @siwane  and @antoinerobin , we concluded that this else statement calling legacy code is the only way to continue being able to run the updaters.